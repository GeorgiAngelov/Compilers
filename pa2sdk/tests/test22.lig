function f() {
      x = 5;
      f().a = 5;
      f()[2] = 5;
      {a = 1}.a = 5;
      [1, 2, 3][0] = 5;

      if (a > b) { x = 1; f(); }
      if (f() > f()) { x = 1; f(); } else { x = 2; f(); }
      while (f()) { }
      for (x = f() to f()) { }

      return (f() + 2);

      return ;
}
