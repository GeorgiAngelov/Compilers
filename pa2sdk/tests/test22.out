Genuine Liger: yes
Function f:
   Arity: 0
   Defined: yes
   Calls: 11
